pkg/prom/query.go

```diff
@@ -157,46 +166,68 @@ func runQuery(query string, ctx *Context, resCh QueryResultsChan, profileLabel s
 	resCh <- results
 }
 
-func (ctx *Context) query(query string) (interface{}, prometheus.Warnings, error) {
+// RawQuery is a direct query to the prometheus client and returns the body of the response
+func (ctx *Context) RawQuery(query string) ([]byte, error) {
```

Comment:
`RawQuery` and `RawQueryRange` were required for proxying queries from the frontend while also piping the queries through our request queue.